File: CodeBattleJava/src/main/java/ru/codebattle/client/api/SnakeAction.java
package ru.codebattle.client.api;

import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.ToString;

@RequiredArgsConstructor
@Getter
public class SnakeAction {
  private static final String ACT_COMMAND_PREFIX = "ACT,";

  private final boolean act;
  private final Direction direction;

  @Override
  public String toString() {
    var cmd = act ? ACT_COMMAND_PREFIX : "";
    return cmd + direction.toString();
  }
}

File: SnakeBattleNet/SnakeBattleApi/SnakeAction.cs
namespace SnakeBattle.Api
{
    public class SnakeAction
    {
        private const string ACT_COMMAND_PREFIX = "ACT,";

        private readonly bool _act;
        private readonly Direction _direction;

        public SnakeAction(bool act, Direction direction)
        {
            _act = act;
            _direction = direction;
        }

        public override string ToString()
        {
            var cmd = _act ? ACT_COMMAND_PREFIX : string.Empty;
            return cmd + _direction;
        }
    }
}
